client/router_update.go

```diff
@@ -1084,6 +1086,56 @@ func (cli *VanClient) RouterUpdateAnnotations(ctx context.Context, settings *cor
 	return updated, nil
 }
 
+func sortHostAliases(hostAliases []corev1.HostAlias) []corev1.HostAlias {
+	sort.Slice(hostAliases, func(i, j int) bool {
+		return hostAliases[i].IP < hostAliases[j].IP
+	})
+
+	for _, hn := range hostAliases {
+		sort.Slice(hn.Hostnames, func(i, j int) bool {
+			return hn.Hostnames[i] < hn.Hostnames[j]
+		})
+	}
+	return hostAliases
+}
+
+func hostAliasesEqual(new, old []corev1.HostAlias) bool {
+	if len(new) != len(old) {
+		return false
+	}
+	return cmp.Equal(sortHostAliases(new), sortHostAliases(old))
+}
+
+func (cli *VanClient) updateHostAliasesOnDeployment(ctx context.Context, namespace string, name string, hostAliases []corev1.HostAlias) (bool, error) {
+	deployment, err := cli.KubeClient.AppsV1().Deployments(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	if !hostAliasesEqual(hostAliases, deployment.Spec.Template.Spec.HostAliases) {
+		deployment.Spec.Template.Spec.HostAliases = hostAliases
+		_, err = cli.KubeClient.AppsV1().Deployments(namespace).Update(ctx, deployment, metav1.UpdateOptions{})
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func (cli *VanClient) RouterUpdateHostAliases(ctx context.Context, token *corev1.Secret) (bool, error) {
+	hostAliases, err := cli.GetRouterHostAliasesSpecFromTokens(ctx)
```

Comment:
This is not working when the `site-controller` is deployed watching ALL namespaces.
You need `GetRouterHostAliasesSpecFromTokens` to use the token's namespace for that, instead of relying on the cli's namespace.

Reply:
Thanks for the review and spotting it, I wasn't even aware of this. The latest commit fixes it.